Libraries/LibWeb/XHR/FormData.cpp

```diff
@@ -18,17 +21,38 @@ namespace Web::XHR {
 GC_DEFINE_ALLOCATOR(FormData);
 
 // https://xhr.spec.whatwg.org/#dom-formdata
-WebIDL::ExceptionOr<GC::Ref<FormData>> FormData::construct_impl(JS::Realm& realm, GC::Ptr<HTML::HTMLFormElement> form)
+WebIDL::ExceptionOr<GC::Ref<FormData>> FormData::construct_impl(JS::Realm& realm, GC::Ptr<HTML::HTMLFormElement> form, Optional<GC::Ptr<HTML::HTMLElement>> submitter)
 {
     Vector<FormDataEntry> list;
     // 1. If form is given, then:
     if (form) {
-        // 1. Let list be the result of constructing the entry list for form.
+        // 1. If submitter is non-null, then:
+        if (submitter.has_value() && submitter.value()) {
+            auto& submitter_element = submitter.value();
+
+            // 1. If submitter is not a submit button, then throw a TypeError.
+            bool is_image_button = submitter_element->is_html_input_element() && dynamic_cast<HTML::HTMLInputElement*>(submitter_element.ptr())->type_state() == HTML::HTMLInputElement::TypeAttributeState::ImageButton;
+            if (!submitter_element->is_html_button_element() && !is_image_button) {
```

Comment:
It isn't obvious why there is a check for an `ImageButton` here since the spec step doesn't mention it.
Maybe worth an AD-HOC or NOTE comment?

Reply:
ImageButton can be a submit button, but a comment should be a good idea

Reply:
`FormAssociatedElement` has a `is_submit_button()` method on it. You should be able to use that here.